src/withAuthUser.js

```diff
@@ -124,48 +127,34 @@ const withAuthUser =
           : true)
 
       const router = useRouter()
+      const routeToDestination = useCallback(
+        ({ basePath, destination }) => {
+          if (basePath === false) {
+            window.location.assign(destination)
+          } else {
+            router.replace(destination)
+          }
+        },
+        [router]
+      )
       const redirectToApp = useCallback(() => {
         logDebug('Redirecting to app.')
-        const appRedirectDestination = appPageURL || getConfig().appPageURL
-        if (!appRedirectDestination) {
-          throw new Error(
-            'The "appPageURL" config setting must be set when using `REDIRECT_TO_APP`.'
-          )
-        }
+        const destination = getRedirectToAppDestination({
+          AuthUser,
+          options: withAuthUserConfig,
```

Comment:
Please explicitly pass only the properties that are used (construct a new object or name each property).